xgi/classes/reportviews.py

```diff
@@ -225,36 +225,72 @@ def filterby(self, stat, val, mode="eq"):
         elif mode == "between":
             bunch = [node for node in self if val[0] <= values[node] <= val[1]]
         else:
-            raise ValueError(f"Unrecognized mode {mode}")
+            raise ValueError(
+                f"Unrecognized mode {mode}. mode must be one of 'eq', 'neq', 'lt', 'gt', 'leq', 'geq', or 'between'."
+            )
         return type(self).from_view(self, bunch)
 
-    def filterby_attr(self, attr, val, mode="eq"):
+    def filterby_attr(self, attr, val, mode="eq", default=None):
```

Comment:
```suggestion
    def filterby_attr(self, attr, val, mode="eq", missing=None):
```

Reply:
I liked your suggestion to match the interface so I implemented it now instead of adding it as an issue